x-pack/platform/plugins/shared/security/server/saved_objects/access_control_service.ts

```diff
@@ -147,6 +149,23 @@ export class AccessControlService {
         }
       }
     }
+
+    for (const type of typesRequiringRbac) {
+      const typeAuth = typeMap.get(type);
+      const rbacAuth = typeAuth?.['update' as A];
+      if (!rbacAuth) {
```

Comment:
Nit/Question: This check looks very similar for the same types as the block above, could it be split into a semantically named helper function and used in both places?